engine/src/main/java/org/terasology/config/flexible/internal/SettingBuilder.java

```diff
@@ -66,8 +55,16 @@
     SettingBuilder<T> description(String description);
 
     /**
-     * Builds the {@link Setting} with the components that have already been specified and
-     * returns it.
+     * Specfies the system property of the {@link Setting} being created.
+     *
+     * @param systemProperty This system property which will rewrite value.
+     * @param converter converter for systemProperty: String -> T
```

Comment:
**nitpick (non-blocking):** To me _converter_ often implies that the conversion can work in both directions, and that it can be between arbitrary types. In this case (converting from `String -> T`) I'd call it a _decoder_ function (the dual function `T -> String` would be the encoder).

Reply:
I don't like idea to write something in sysProps at runtime.
It are going to problems :D I saw systems, where general config source - is  sysProps. 
then we catch problem like CoreRegistry + parallel tests.. it was worst.